cmd/osm-controller/osm-controller.go

```diff
@@ -214,7 +214,9 @@ func main() {
 		cfg,
 		endpointsProviders...)
 
-	proxyRegistry := registry.NewProxyRegistry(&registry.KubeProxyServiceMapper{KubeController: kubernetesClient})
+	proxyMapper := registry.NewKubeProxyServiceMapper(kubernetesClient)
+	go proxyMapper.Run(stop)
+	proxyRegistry := registry.NewProxyRegistry(proxyMapper)
```

Comment:
Hey @nojnhuh there have been a few different approaches proposed to refactor this code:
1. proxy-service mapper in pkg/envoy
2. catalog2 by @eduser25 
3. XDS convertor shim to convert platform-specific (ex. k8s) constructs to XDS config (proposed by me)

What do you think about evaluating these proposals to iron out the gaps in our existing implementation?

Reply:
Yes, I'm happy to address those.

> 1. proxy-service mapper in pkg/envoy

Moving this is outside the scope of these particular changes, but I understand referencing k8s in pkg/envoy isn't the best. This may depend on 3., but where do you think is an "ideal" place for this? Or at least a "good enough for now" place?

> 2. catalog2 by @eduser25

If we agree it's worth keeping this functionality factored out of the catalog, then catalog2 shouldn't conflict with this at all. This would continue to live on separately but following a similar pattern in how it works.

> 3. XDS convertor shim to convert platform-specific (ex. k8s) constructs to XDS config (proposed by me)

Let me take another look at where you first mentioned this, but I may need some help understanding exactly what this would look like.

Reply:
> Yes, I'm happy to address those.
> 
> > 1. proxy-service mapper in pkg/envoy
> 
> Moving this is outside the scope of these particular changes, but I understand referencing k8s in pkg/envoy isn't the best. This may depend on 3., but where do you think is an "ideal" place for this? Or at least a "good enough for now" place?
> 
I would put this in MeshCatalog as a start so we don't leak k8s stuff into pkg/envoy.

> > 1. catalog2 by @eduser25
> 
> If we agree it's worth keeping this functionality factored out of the catalog, then catalog2 shouldn't conflict with this at all. This would continue to live on separately but following a similar pattern in how it works.
> 
> > 1. XDS convertor shim to convert platform-specific (ex. k8s) constructs to XDS config (proposed by me)
> 
> Let me take another look at where you first mentioned this, but I may need some help understanding exactly what this would look like.

The XDS shim is a proposal I have in mind (no doc yet just offline brainstorming). I think it would be best to collaborate and align the different proposals over a doc. @draychev, @eduser25, and I are willing to collaborate on this effort to figure out what our optimization requirements are.

Reply:
For the record - as of this writing - this PR is the only proposal we have. (This is already a great step forward in cleaning up Mesh Catalog)
We have not seen material proposals on catalog2 or XDS shim.